internal/datastore/spanner/gc.go

```diff
@@ -0,0 +1,66 @@
+package spanner
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"cloud.google.com/go/spanner"
+	sq "github.com/Masterminds/squirrel"
+	"github.com/go-co-op/gocron"
+	"github.com/rs/zerolog/log"
+)
+
+type cancelFunc func()
+
+var noopCancelFunc cancelFunc = func() {}
+
+func (sd spannerDatastore) runGC() cancelFunc {
+	if sd.config.gcInterval <= 0 {
+		log.Info().Stringer("interval", sd.config.gcInterval).Msg("garbage collection: disabled")
+		return noopCancelFunc
+	}
+
+	log.Info().Stringer("interval", sd.config.gcInterval).Msg("garbage collection: starting")
+
+	s := gocron.NewScheduler(time.UTC)
+	var numRemoved int64
+	_, err := s.Every(sd.config.gcInterval).Do(func() {
```

Comment:
We might want some jitter here, or potentially even have the instances coordinate via spanner

Reply:
Add a random jitter for now and a comment to make it coordinate?

Reply:
Added jitter, I'll make a GH issue for coordination instead of a comment